defaults.go

```diff
@@ -150,13 +148,10 @@ func setField(field reflect.Value, defaultVal string) error {
 		setField(field.Elem(), defaultVal)
 		callSetter(field.Interface())
 	case reflect.Struct:
-		ref := reflect.New(field.Type())
-		ref.Elem().Set(field)
-		if err := Set(ref.Interface()); err != nil {
+		if err := Set(field.Addr().Interface()); err != nil {
 			return err
 		}
-		callSetter(ref.Interface())
-		field.Set(ref.Elem())
+		callSetter(field.Addr().Interface())
```

Comment:
It seems to be a redundant call as you're already doing it in `Set`.
Either one of them should be removed. Perhaps the latter.